tls/s2n_early_data.c

```diff
@@ -167,6 +167,18 @@ S2N_RESULT s2n_early_data_get_server_max_size(struct s2n_connection *conn, uint3
     return S2N_RESULT_OK;
 }
 
+int s2n_connection_set_server_early_data_context(struct s2n_connection *conn, const uint8_t *context, uint16_t context_size)
+{
+    POSIX_ENSURE_REF(conn);
+    if (context_size > 0) {
+        POSIX_ENSURE_REF(context);
+    }
```

Comment:
If ```context_size <= 0``` we are handing out pointers to our caller, which goes against the following S2N Guideline:

-  _Prefer copying data in and out of s2n over taking and handing out pointers to internal memory with a limited lifetime._

We must `POSIX_ENSURE_REF(context)` and `POSIX_ENSURE_GT(context_size, 0);` instead here.